### ISSUE TITLE
Support for competing risks simulation

### ISSUE THREAD

**Issue by adibender** [adibender/pammtools#106]
- [ ] simulation 
  - [ ] interface 
  - [ ] basic 
  - [ ] time-varying effects 
  - [ ] time-dependent covariates
  - [ ] cumulative effects

see #105

**Comment by pkopper** [comment]
We should discuss how the existing simulation may be adapted to facilitate the new functions.

**Comment by fabian-s** [comment]
@adibender : if I read this correctly you'd suggest a new function `sim_pexp_cr` that deals with CR complications and that gets called from `sim_pexp`, which is then simply a wrapper for calling either the new `sim_pexp_cr` or the "old"  `sim_pexp` (renamed as `sim_pexp_single`, or similar, see below), depending on the formula structure?

OT: probably need to agree on a global naming scheme rule to differentiate CR functionality from "classic" functionality? suggestion: call one "<BLA>_cr"  and the other "<BLA>_1r" (one risk)?

**Comment by fabian-s** [comment]
.. and internallly, `sim_pexp_cr` will probably just a wrapper around calling (parts of) `sim_pexp_single` for each of the competing risk processes and then using the minimum of the generated event times as the actualized risk/event type, correct?

**Comment by adibender** [comment]
Right, I think it would be best to refactor the existing `sim_pexp` such that

1. there is a function that constructs the linear predictor given specification of the RHS of the input formula (and data)
2. within `sim_pexp` we calculate one linear predictor/hazard for each part of the RHS of the input formula, i.e. for `~ 0.2 * x1 | 0.1 * x2` we calculate two linear predictors (that requieres to fix #88 first)
That could be done by looping through the number of RHS arguments and calling the function from 1. 
3. If the List returned in 2. is longer than one call `sim_pexp_cr`, otherwise call `sim_pexp_1r`
Within `sim_pexp_cr` it proceeds similar to what I already implemented in the `competing-risks` branch, but could probably be improved upon.

**Comment by pkopper** [comment]
Thank you. I think we are on the same page w.r.t. this.
